### PR TITLE
[ASSIST-389] Engagement Timeline

### PR DESCRIPTION
**Pull request by dgoetzit**
This PR introduces a `timeline` module which will make it relatively easy for us to make any model "timelineable".

### CODE REVIEW

File: tailwind.config.js [hunk 1]
```diff
@@ -25,4 +25,5 @@ export default {
     plugins: [
         require('flowbite/plugin')
     ],
-}
+    plugins: [forms, typography],
+};
```
Comment (Orrison) [inline tailwind.config.js:29]: Why are we removing `flowbite` here?

**Review by Orrison — COMMENTED** [review]
Looks good, just a few comments and questions. There also is currently a merge conflict.

File: app-modules/engagement/src/Models/Engagement.php [hunk 2]
```diff
@@ -33,6 +39,40 @@ class Engagement extends BaseModel implements Auditable, CanTriggerAutoSubscript
...
+    public function modalViewAction(): ViewAction
+    {
+        return EngagementViewAction::make()->record($this);
+    }
+
+    public static function getTimeline(Model $forModel): Collection
+    {
+        return $forModel->engagements()->with(['deliverables', 'batch'])->get();
+    }
+
```
[the remark this answers is not in the document]
Comment (dgoetzit) [reply app-modules/engagement/src/Models/Engagement.php:75]: Yeah I think that's a good idea 👍

File: app-modules/timeline/src/Filament/Pages/Timeline.php [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        return $this->currentRecordToView->modalViewAction();
+    }
+
+    protected function authorizeAccess(): void
+    {
+        static::authorizeResourceAccess();
+
+        // TODO We also need to check access for the other entities that are going to be included in the timeline
+        // We probably just need to establish that the user can view any of a model, but might need to be more specific
+        abort_unless(static::getResource()::canView($this->getRecord()), 403);
```
[the remark this answers is not in the document]
Comment (dgoetzit) [reply app-modules/timeline/src/Filament/Pages/Timeline.php:80]: Yes, https://engage.canyongbs.com/workgroups/group/3/tasks/task/view/493/

File: tailwind.config.js (hunk 1 above)
Comment (dgoetzit) [reply tailwind.config.js:29]: Ahh whoops - oversight from merge.

File: app-modules/engagement/src/Models/Engagement.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (dgoetzit) [reply app-modules/engagement/src/Models/Engagement.php:53]: Let me know what you think of the latest approach - utilizes a custom `{Model}Timeline` class for each model that provides a `Timeline`.

**Review by Orrison — APPROVED** [review]
Looks great! I think there is just a merge conflict left

**Review by Orrison — APPROVED** [review]

**Pull request merged**